webapp/src/modules/order/sagas.ts

```diff
@@ -57,17 +64,33 @@ function* handleExecuteOrderRequest(action: ExecuteOrderRequestAction) {
   const { order, nft, fingerprint } = action.payload
   try {
     if (
-      nft.contractAddress !== order.contractAddress &&
+      nft.contractAddress !== order.contractAddress ||
       nft.tokenId !== order.tokenId
     ) {
       throw new Error('The order does not match the NFT')
     }
-    const { orderService } = VendorFactory.build(nft.vendor)
+    const { orderService }: Vendor<VendorName> = yield call(
+      [VendorFactory, 'build'],
+      nft.vendor
+    )
 
     const wallet: ReturnType<typeof getWallet> = yield select(getWallet)
-    const txHash: string = yield call(() =>
-      orderService.execute(wallet, nft, order, fingerprint)
+    const txHash: string = yield call(
+      [orderService, 'execute'],
+      wallet,
+      nft,
+      order,
+      fingerprint
     )
+    if (nft.openRentalId) {
```

Comment:
Before start cancelling the rental listing, I think we should wait until the tx is confirmed. The execute only triggers the tx so this would inmediately start trying to cancel the listing even is the tx is not confirmed yet. Wdyt? 

```suggestion
    yield call(waitForTx, txHash)
    if (nft.openRentalId) {
```

Reply:
It would be great to use the `waitForTx` helper but the transaction is dispatched to the state in the `executeOrderSuccess` so we would have to make more changes in the code to make it work.